### ISSUE TITLE
verdi code setup should warn when setting up a code with existing label

### ISSUE THREAD

**Issue by ltalirz** [aiidateam/aiida-core#3303]
Currently, `verdi code setup` will happily set up codes with the same label multiple times.
It is fine that this is allowed in principle, but in interactive mode, a user should be warned when doing this.

The simplest might be something like:
```
$ verdi code setup
Info: enter "?" for help
Info: enter "!" to ignore the default and set no value
Label: cp
Description []: asdf
Default calculation input pluginquantumespresso.pw  
Installed on target computer? [True]: 
Computer: localhost
Warning: Code 'cp@localhost' already exists.
Remote absolute path:
```
If we want to be more fancy, we could add a prompt
```
Choose a different label [Y/n]: 
```

**Comment by ltalirz** [comment]
Along the same lines: when importing an archive file, it would make sense to relabel codes in the archive if a code of the same label already exists in the database (analogous to what is done for computers).
Otherwise, imports are likely to break scripts that rely on loading codes by label (which is a common practise).

**Comment by mbercx** [comment]
Just out of curiosity: Would it not be better to disallow two codes with the same label altogether?

**Comment by ltalirz** [comment]
> Would it not be better to disallow two codes with the same label altogether?

Once we implement the renaming for `verdi import`, we're almost there in practice.
One problem might be that codes are represented as nodes, so I guess one cannot easily enforce this at the database schema level (while computers are not).

**Comment by ramirezfranciscof** [comment]
>  One problem might be that codes are represented as nodes, so I guess one cannot easily enforce uniqueness at the database schema level (while computers are not).

But we could enforce uniqueness at the level of the `node:code` subclass no? Forcing it to check the DB for existing labels before creating (init?) and/or when changing the label property.

**Comment by greschd** [comment]
> Just out of curiosity: Would it not be better to disallow two codes with the same label altogether?

In my opinion the code label doesn't need to be unique. For context, when testing a code across variations of our infrastructure I would distinguish them only by the computer. Of course it would be possible to add a `-<computer_name>` to the code label, but that seems silly if there's already the `@<computer_name>` syntax.

**Comment by ltalirz** [comment]
Yes, my comment seemed to imply that I was thinking of uniqueness at the code label only, but I agree that this discussion should focus on whether `code-label@computer-label` should be unique (not the code label alone, as @greschd rightfully points out)

**Comment by janssenhenning** [comment]
Hello Aiida team,
I'm one of the developers of the aiida-fleur plugin and I want to also start making some contributions to aiida-core.
If this issue is still relevant I would be happy to work on it

**Comment by mbercx** [comment]
Hi @janssenhenning! 👋  Welcome, happy to hear you want to work on this!

I definitely think it is still relevant! I guess we just have to agree on how to proceed. To continue the discussion, let me reply to @ltalirz's comment.

> Yes, my comment seemed to imply that I was thinking of uniqueness for the code label only, but I agree that this discussion should focus on whether `code-label@computer-label` should be unique (not the code label alone, as @greschd rightfully points out)

Indeed, I agree that the code label _alone_ should not need to be unique. However, I think we should enforce only having one `Code` with a specific label on each computer, so we'll never have 3 `add@localhost` code installed (as definitely happened during the tutorial).

**Comment by janssenhenning** [comment]
I started trying to implement the mentioned warning mechanism for ``verdi code setup`` and ``verdi code duplicate`` for duplicate ``code_label@computer_label`` on my fork. I encountered a couple of things I'm not sure about:

- What should happen when the command is called in non-interactive mode? Should the command exit with an error or should it be allowed?
- I did this by adding an additional callback to the computer option, which then triggers a new label option if the label combination already exists. Doing this leads to the additional callback being called twice (I already opened an issue for this #5023)
- Is there a way to create a local code in the test system with a fixture? With the ``aiida_local_code_factory`` I can only create remote codes on ``localhost``
